lib/ui/platform_dispatcher.dart

```diff
@@ -0,0 +1,1586 @@
+  /// original callback is called in the zone it was registered in.
+  static PlatformMessageResponseCallback? _zonedPlatformMessageResponseCallback(
+    PlatformMessageResponseCallback? callback,
+  ) {
+    if (callback == null) {
+      return null;
+    }
+
+    // Store the zone in which the callback is being registered.
+    final Zone registrationZone = Zone.current;
+
+    return (ByteData? data) {
+      registrationZone.runUnaryGuarded(callback, data);
+    };
+  }
+
+  // Called from the engine, via hooks.dart
+  void _dispatchPlatformMessage(String name, ByteData? data, int responseId) {
+    if (name == ChannelBuffers.kControlChannelName) {
+      try {
+        channelBuffers.handleMessage(data!);
+      } catch (ex) {
+        _printDebug('Message to "$name" caused exception $ex');
+      } finally {
+        _respondToPlatformMessage(responseId, null);
+      }
+    } else if (onPlatformMessage != null) {
+      _invoke3<String, ByteData?, PlatformMessageResponseCallback>(
+        onPlatformMessage,
+        _onPlatformMessageZone,
+        name,
+        data,
+            (ByteData? responseData) {
+          _respondToPlatformMessage(responseId, responseData);
+        },
+      );
+    } else {
+      channelBuffers.push(name, data, (ByteData? responseData) {
+        _respondToPlatformMessage(responseId, responseData);
+      });
+    }
+  }
+
+  /// Set the debug name associated with this platform dispatcher's root
+  /// isolate.
+  ///
+  /// Normally debug names are automatically generated from the Dart port, entry
+  /// point, and source file. For example: `main.dart$main-1234`.
+  ///
+  /// This can be combined with flutter tools `--isolate-filter` flag to debug
+  /// specific root isolates. For example: `flutter attach --isolate-filter=[name]`.
+  /// Note that this does not rename any child isolates of the root.
+  void setIsolateDebugName(String name) native 'PlatformConfiguration_setIsolateDebugName';
+
+  /// The embedder can specify data that the isolate can request synchronously
+  /// on launch. This accessor fetches that data.
+  ///
+  /// This data is persistent for the duration of the Flutter application and is
+  /// available even after isolate restarts. Because of this lifecycle, the size
+  /// of this data must be kept to a minimum.
+  ///
+  /// For asynchronous communication between the embedder and isolate, a
+  /// platform channel may be used.
+  ByteData? getPersistentIsolateData() native 'PlatformConfiguration_getPersistentIsolateData';
+
+  /// Requests that, at the next appropriate opportunity, the [onBeginFrame] and
+  /// [onDrawFrame] callbacks be invoked.
+  ///
+  /// See also:
+  ///
+  ///  * [SchedulerBinding], the Flutter framework class which manages the
+  ///    scheduling of frames.
+  void scheduleFrame() native 'PlatformConfiguration_scheduleFrame';
+
+  /// Updates the application's rendering on the GPU with the newly provided
```

Comment:
If I remember correctly, FlutterView also has a render function. Should this doc mention that one? Maybe even make a recommendation which one to call?